### ISSUE TITLE
Add modules CI checks

### ISSUE THREAD

**Issue by msimberg** [STEllAR-GROUP/hpx#4497]
We can add the following checks:

- All header files need to be listed in `CMakeLists.txt`.
- All direct dependencies should be listed in `CMakeLists.txt` (and conversely no transitive dependencies need to be listed), this should be doable when module headers follow the `hpx/<modulename>` naming convention.

**Comment by hkaiser** [comment]
@aurianer I think we should list all direct dependencies (i.e. modules for which a header is included) even if those would be pulled in transitively.

**Comment by aurianer** [comment]
I agree but I prefer to wait until the modules are pretty much in place, because some of the modules have complicated dependencies just because other modules are not yet formed, I'll update the script as soon as they are all approximately in place :)

**Comment by msimberg** [comment]
@aurianer this is all implemented in the current checks, right? (Except for the "no transitive dependencies should be listed"-check, but that is trickier anyway...)
